__tests__/services/CleanupService.spec.ts

```diff
@@ -104,116 +150,644 @@ const mockCookie: CookiePropertiesCleanup = {
+
+        expect(ffResult!.cachedResults.recentlyCleaned).toBe(5);
+        expect(ffResult!.setOfDeletedDomainCookies).toHaveLength(5);
+        expect(ffResult!.setOfDeletedDomainCookies).toEqual(['test.com','google.com', 'yahoo.com', 'sub.domain.com', 'github.com']);
+      });
+
+      it('Regular clean, include open tabs (Firefox).', async () => {
+        const ffResult = await cleanCookiesOperation(firefoxState, {...cleanupProperties, ignoreOpenTabs: true});
+        expect(global.browser.cookies.remove).toHaveBeenCalledTimes(3);
+
+        expect(ffResult!.cachedResults.recentlyCleaned).toBe(3);
+        expect(ffResult!.setOfDeletedDomainCookies.length).toBe(3);
+        expect(ffResult!.setOfDeletedDomainCookies).toEqual(['test.com','yahoo.com', 'sub.domain.com']);
+      });
+
+      it('Regular clean, exclude open tabs (Firefox), with only cookies in open tabs/whitelist.', async () => {
+        when(global.browser.cookies.getAll)
+          .calledWith({storeId: 'firefox-default'})
+          .mockResolvedValue([googleCookie, youtubeCookie] as never);
+        const ffResult = await cleanCookiesOperation(firefoxState, cleanupProperties);
+        expect(global.browser.cookies.remove).not.toHaveBeenCalled();
+        expect(ffResult!.cachedResults.recentlyCleaned).toBe(0);
+        expect(ffResult!.setOfDeletedDomainCookies.length).toBe(0);
+        expect(ffResult!.setOfDeletedDomainCookies).toEqual([]);
+      });
+
+      it('Regular clean, exclude open tabs (Firefox) with debugMode enabled', async () => {
+        await cleanCookiesOperation(firefoxDebugState, cleanupProperties);
+        expect(spyLib.cadLog).toHaveBeenCalledTimes(40);
+      });
+
+      it('Regular clean, exclude open tabs (Firefox) with debugMode enabled to include errors during browser.cookies.getAll', async () => {
+        when(global.browser.cookies.getAll)
+          .calledWith(expect.any(Object))
+          .mockResolvedValueOnce([] as never)
+          .mockRejectedValue(new Error('test') as never);
+        await cleanCookiesOperation(firefoxDebugState, cleanupProperties);
+        expect(global.browser.cookies.getAll).toHaveBeenCalledTimes(3);
+        // one for each storeId: 'default', 'firefox-default'
+        expect(spyLib.cadLog).toHaveBeenCalledTimes(2);
+      });
+
+      it('Regular clean, exclude open tabs (Firefox) to include errors during cleanCookies / browser.cookies.remove (via cadLog)', async () => {
+        when(global.browser.cookies.remove)
+          .calledWith(expect.any(Object))
+          .mockRejectedValue(new Error('test') as never);
+        await cleanCookiesOperation(firefoxState, cleanupProperties);
+        // isFirstPartyIsolate plus one for each storeId: 'default', 'firefox-default'
+        expect(global.browser.cookies.getAll).toHaveBeenCalledTimes(3);
+        // this error always call cadLog even if debugMode is disabled.
+        expect(spyLib.cadLog).toHaveBeenCalledTimes(1);
```

Comment:
I'm not sure if it's necessary to spy the cadlog in here and other places.

Reply:
With coverage, at least one test is needed to cover the functions called, including cadLog, depending on how the path to it is given.